### DISCUSSION TITLE
Kyocera Ecosys M2135 ArgumentException parameter is not valid after scan is done

### DISCUSSION THREAD

**Discussion by seriousguymaybe** [cyanfish/naps2#229] [Support]
Hi Guys,

PS. Why are you using discussion when have "issue management system" on GitHub?

**To Reproduce**
1. Press scan icon
2. Start scanning from UI of Kyocera
3. Scan is done you get error message

**Expected behavior**
To get scanned document on main work area

**Screenshots**
![image](https://github.com/cyanfish/naps2/assets/42060865/6bbaadfb-b3bd-43da-a61b-0a91b6d9a7b3)

**Desktop (please complete the following information):**
Windows 10 19045.3693   22H2

**Was working on older version 6.1.2 without problem**

Error message:
System.ArgumentException: Parameter is not valid.
   at new System.Drawing.Bitmap(int width, int height, PixelFormat format)
   at IMemoryImage NAPS2.Images.Gdi.GdiImageContext.Create(int width, int height, ImagePixelFormat pixelFormat)
   at void NAPS2.Scan.Internal.Twain.TwainImageProcessor.MemoryBufferTransferred(TwainMemoryBuffer memoryBuffer)
   at async Task NAPS2.Remoting.Worker.WorkerServiceAdapter.TwainScan(ScanOptions options, CancellationToken cancelToken, ITwainEvents twainEvents)
   at async Task NAPS2.Scan.Internal.Twain.RemoteTwainSessionController.StartScan(ScanOptions options, ITwainEvents twainEvents, CancellationToken cancelToken)
   at async Task NAPS2.Scan.Internal.Twain.TwainScanDriver.Scan(ScanOptions options, CancellationToken cancelToken, IScanEvents scanEvents, Action<IMemoryImage> callback)+(?) => { }
   at async Task NAPS2.Scan.Internal.RemoteScanController.Scan(ScanOptions options, CancellationToken cancelToken, IScanEvents scanEvents, Action<ProcessedImage, PostProcessingContext> callback)
   at async IAsyncEnumerable<ProcessedImage> NAPS2.Scan.ScanController.Scan(ScanOptions options, CancellationToken cancelToken)+(?) => { }

Hope it helps

**Comment by cyanfish** [comment]
Can you follow [these instructions](https://www.naps2.com/doc/troubleshooting#debug-log) to collect debug logs?

In your profile settings, under Advanced, you can try selecting "Native Transfer" under Twain Implementation.

Re: issues, a lot of scanner support questions end up unsolvable or the creator doesn't respond, and I prefer issues to be things that can be definitively resolved and closed.

**Reply by seriousguymaybe** [reply]
Hi,

"Native Transfer" option is already used.
Bellow are "debuglog" and "errolog" files.
GitHub don't allow xml files as upload so i pasted content bellow.
Hope this will help you more.
If i find some time i can debug NAPS2 in VS and give you more glimpse what is wrong.


[debuglog.txt](https://github.com/cyanfish/naps2/files/13449981/debuglog.txt)
[errorlog.txt](https://github.com/cyanfish/naps2/files/13449982/errorlog.txt)

Config XML

`<?xml version="1.0" encoding="utf-8"?>
<UserConfig>
  <KeyboardShortcuts />
  <HasBeenRun>true</HasBeenRun>
  <FirstRunDate>2023-11-23T13:40:24.4994409+01:00</FirstRunDate>
  <Version>3</Version>
  <EnableDebugLogging>true</EnableDebugLogging>
  <FormStates>
    <FormState>
      <Name>AboutForm</Name>
      <Location>
        <X>932</X>
        <Y>541</Y>
      </Location>
      <Size>
        <Width>432</Width>
        <Height>208</Height>
      </Size>
      <Maximized>false</Maximized>
    </FormState>
    <FormState>
      <Name>EditProfileForm</Name>
      <Location>
        <X>937</X>
        <Y>431</Y>
      </Location>
      <Size>
        <Width>422</Width>
        <Height>426</Height>
      </Size>
      <Maximized>false</Maximized>
    </FormState>
    <FormState>
      <Name>ProfilesForm</Name>
      <Location>
        <X>869</X>
        <Y>567</Y>
      </Location>
      <Size>
        <Width>558</Width>
        <Height>155</Height>
      </Size>
      <Maximized>false</Maximized>
    </FormState>
    <FormState>
      <Name>ErrorForm</Name>
      <Location>
        <X>941</X>
        <Y>603</Y>
      </Location>
      <Size>
        <Width>413</Width>
        <Height>209</Height>
      </Size>
      <Maximized>false</Maximized>
    </FormState>
  </FormStates>
  <BatchSettings />
</UserConfig>
`

Profiles.xml

`<?xml version="1.0" encoding="utf-8"?>
<ArrayOfScanProfile>
  <ScanProfile>
    <Version>2</Version>
    <Device>
      <ID>ECOSYS M2135dn (1)</ID>
      <Name>ECOSYS M2135dn (1)</Name>
    </Device>
    <DriverName>twain</DriverName>
    <DisplayName>ECOSYS M2135dn (1)</DisplayName>
    <IconID>0</IconID>
    <MaxQuality>false</MaxQuality>
    <IsDefault>true</IsDefault>
    <UseNativeUI>true</UseNativeUI>
    <AfterScanScale>OneToOne</AfterScanScale>
    <Brightness>0</Brightness>
    <Contrast>0</Contrast>
    <BitDepth>C24Bit</BitDepth>
    <PageAlign>Right</PageAlign>
    <PageSize>A4</PageSize>
    <CustomPageSizeName p3:nil="true" xmlns:p3="http://www.w3.org/2001/XMLSchema-instance" />
    <CustomPageSize p3:nil="true" xmlns:p3="http://www.w3.org/2001/XMLSchema-instance" />
    <Resolution>Dpi100</Resolution>
    <PaperSource>Glass</PaperSource>
    <EnableAutoSave>false</EnableAutoSave>
    <AutoSaveSettings>
      <FilePath></FilePath>
      <PromptForFilePath>false</PromptForFilePath>
      <ClearImagesAfterSaving>false</ClearImagesAfterSaving>
      <Separator>FilePerPage</Separator>
    </AutoSaveSettings>
    <Quality>75</Quality>
    <AutoDeskew>false</AutoDeskew>
    <BrightnessContrastAfterScan>false</BrightnessContrastAfterScan>
    <ForcePageSize>false</ForcePageSize>
    <ForcePageSizeCrop>false</ForcePageSizeCrop>
    <TwainImpl>Default</TwainImpl>
    <ExcludeBlankPages>false</ExcludeBlankPages>
    <BlankPageWhiteThreshold>70</BlankPageWhiteThreshold>
    <BlankPageCoverageThreshold>25</BlankPageCoverageThreshold>
    <WiaOffsetWidth>false</WiaOffsetWidth>
    <WiaRetryOnFailure>false</WiaRetryOnFailure>
    <WiaDelayBetweenScans>false</WiaDelayBetweenScans>
    <WiaDelayBetweenScansSeconds>2</WiaDelayBetweenScansSeconds>
    <WiaVersion>Default</WiaVersion>
    <FlipDuplexedPages>false</FlipDuplexedPages>
    <KeyValueOptions p3:nil="true" xmlns:p3="http://www.w3.org/2001/XMLSchema-instance" />
  </ScanProfile>
</ArrayOfScanProfile>`

**Comment by cyanfish** [comment]
Can you try this test version and see if it makes any difference? If not, can you enable debug logging in it (and collect the log from the Data folder)?

[naps2-7.1.2-testtwain.zip](https://github.com/cyanfish/naps2/files/13452404/naps2-7.1.2-testtwain.zip)

**Reply by seriousguymaybe** [reply]
Tried all advanced modes of scanning.
From native to x64 all are returning error.

Legacy is stuck on waiting for twain forever :-(.

Debug Log added.
[debuglog.txt](https://github.com/cyanfish/naps2/files/13473392/debuglog.txt)

**Reply by cyanfish** [reply]
Is that debuglog from the test version? It shows up in the Data folder in the extracted test version, not the same location as the installed version.

**Reply by seriousguymaybe** [reply]
Debug and Error log from extracted [naps2-7.1.2-testtwain.zip] version in Data subfolder:
[debuglog.txt](https://github.com/cyanfish/naps2/files/13478363/debuglog.txt)
[errorlog.txt](https://github.com/cyanfish/naps2/files/13478370/errorlog.txt)

**Reply by cyanfish** [reply]
Thanks, that's helpful, can you try this one now and see if it makes a difference (also please attach the debuglog from this one too):

[naps2-7.1.2-testtwain2.zip](https://github.com/cyanfish/naps2/files/13484682/naps2-7.1.2-testtwain2.zip)

**Reply by seriousguymaybe** [reply]
Default twain implementation gave nothing....he finishes scanning and no page is generated in main menu

But native transfer scanned the page

![image](https://github.com/cyanfish/naps2/assets/42060865/15fa653d-2068-4036-904d-e491ea1943d4)



[debuglog.txt](https://github.com/cyanfish/naps2/files/13486570/debuglog.txt)

**Reply by seriousguymaybe** [reply]
And one more side note.....app is nice and simple to use but it can be complex for basic users.....
If you make wizard mode for scanning like walkthrough that would increase your user base and touch bottom part of users.

Nice app i personally use it already for long time!

Best regards

**Reply by cyanfish** [reply]
Thanks, one more, can you try this one (with Default twain impl) and attach the debuglog?

[naps2-7.1.2-test3.zip](https://github.com/cyanfish/naps2/files/13495730/naps2-7.1.2-test3.zip)

**Reply by seriousguymaybe** [reply]
Yeah np. 
Working!
[debuglog.txt](https://github.com/cyanfish/naps2/files/13587417/debuglog.txt)
